01-shapefile-attributes.Rmd

```diff
@@ -75,182 +79,292 @@ CSV to Shapefile in R</a>
-
 #view class
-class(x = aoiBoundary)
-class(x = lines)
-class(x = point)
+class(x = aoiBoundary_HARV)
+class(x = lines_HARV)
+class(x = point_HARV)
 
+# x= isn't actually needed; it just specifies which object
 #view features count
-length(x = aoiBoundary)
-length(x = lines)
-length(x = point)
+length(aoiBoundary_HARV)
+length(lines_HARV)
+length(point_HARV)
 
 #view crs - note - this only works with the raster package loaded
-crs(x = aoiBoundary)
-crs(x = lines)
-crs(x = point)
+crs(aoiBoundary_HARV)
+crs(lines_HARV)
+crs(point_HARV)
 
-#view extent
-extent(x = aoiBoundary)
-extent(x = lines)
-extent(x = point)
+#view extent- note - this only works with the raster package loaded
+extent(aoiBoundary_HARV)
+extent(lines_HARV)
+extent(point_HARV)
 
 ```
 
-#Querying Shapefile Attributes
+When we type in the full file name we see the same three types of metadata with
+some additional attribute information.  
+
+```{r view-shapefile-metadata2 }
+
+aoiBoundary_HARV
+lines_HARV
+point_HARV
+
+```
+
+#Query Shapefile Attributes
 Shapefiles often contain an associated database or spreadsheet of values called
 `attributes` that describe the vector features in the shapefile. You can think
 of this like a spreadsheet with rows and columns. Each column in the spreadsheet
-is an individual `Attribute` that describes an object. Shapefile `attributes` 
-include measurements that correspond to the geometry of the shapefile features.
+is an individual `attribute` that describes an object (sometimes called
+variables). Shapefile `attributes` include measurements that correspond to the
+geometry of the shapefile features.
+
+For example, the Roads shapefile (`lines_HARV` object) contains an attribute
+called `TYPE`. Each line in the shapefile has an associated TYPE which describes
+the type of road (road, footpath, boardwalk, etc).  
 
-For example, the `roads` shapefile contains an attribute called `TYPE`. Each
-line in the shapefile has an associated type which describes the type of road, or 
-element in the file. 
+We can look at all of the associated data attributes by printing the contents of
+the `data` slot with `@data`. 
 
+```{r shapefile-attributes}
+#just view the attributes of the data
+(lines_HARV@data)
+```
+
+Bonus: If you are comfortable using `slot()`, it is another method to do the
+same thing. {: .notice1}
 
-You can quickly view all attributes of a spatial object in r using the `@data`. 
-Bonus: If you are comfortable using `slot()` it is another method to do the same
-thing. 
+Using `@data` we can see a table with attribute names across the top and rows
+for each line (or point/ polygon if using those) in the shapefile. The 0-12
+printed in the first column is simply a `row.name`. We can test this by issuing
+the following command to the `R` console: `row.names(aoiBoundary@data)`.
+
+``` {r row-names}
+#What are the row names?
+row.names(lines_HARV@data)
+```
+
+If we want to select something about the attribuates, we can do this by 
```

Comment:
Spelling but also i'm not sure what this sentence means. If we want to select something about the attributes?